etc/cmake/options.cmake

```diff
@@ -89,6 +89,7 @@ ot_option(OT_BORDER_AGENT OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE "border agent")
 ot_option(OT_BORDER_AGENT_ID OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE "create and save border agent ID")
 ot_option(OT_BORDER_ROUTER OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE "border router")
 ot_option(OT_BORDER_ROUTING OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE "border routing")
+ot_option(OT_BORDER_ROUTING_DHCP6_PD OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE "dhcpv6 pd support in border routing")
```

Comment:
This flag actually allows the OT stack to process incoming RA messages and adds corresponding routes by looking at the PIO options. The prefixes in the RA message doesn't necessary being a PD prefix.

So what about use a different option name? For example, `OT_BORDER_ROUTING_ACCEPT_RA` (`ACCEPT_RA` comes from linux sysconfig, and they really enables similar function).

Reply:
I see this marked as resolved. Did we decide to keep the `DHCP6_PD` config name?
I think I may have suggested using this name. Users can tell from it what feature they are getting. From `ACCEPT_RA` users cannot immediately tell the feature they get.

> This flag actually allows the border router to process incoming RA messages and adds corresponding routes by looking at the PIO options. The prefixes in the RA message doesn't necessary being a PD prefix.

I think we always accept and process RAs (handle RIO/PIO) from AIL netif (when `ROUTING_MANAGER` is enabled). We have `otPlatInfraIfRecvIcmp6Nd()` platform API to receive all ICMP6 ND including RA msgs from AIL.

For DHCP6 PD we currently have a new plat API `otPlatBorderRoutingProcessIcmp6Ra()` to receive RAs from **_Thread netif_** (which I think are generated on device itself) and we currently only parse its PIO for DHCP6 PD.

Reply:
With some offline discussion, we also agreed that we should use the same flag to guard some logic that is related to the platform, like otPlatBorderRoutingSetPdEnabled (for telling the platform to enable / disable PD on this device) or other related functions.